core-java-modules/core-java-io-5/src/test/java/com/baeldung/charstreams/CharStreamsUnitTest.java

```diff
@@ -0,0 +1,114 @@
+            fwFile.mkdirs();
+            fwFile.createNewFile();
+        } catch(IOException e) {
+            e.printStackTrace();
+        }
+        
+    }
+
+    @Test
+    public void whenUsingTextData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            pwFile = CharStreams.writingStringToFilePrintWriter(TODO_LIST_PW_FILE , TODO.toString());
+            assertThat(pwFile).hasContent(TODO.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingTextData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            fwFile = CharStreams.writingStringToFileFileWriter(TODO_LIST_FW_FILE, TODO.toString());
+            assertThat(fwFile).hasContent(TODO.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+        
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            pwFile = CharStreams.writingNonStringToFilePrintWriter(TODO_LIST_PW_FILE, TODO_1);
+            assertThat(pwFile).hasContent(TODO_1.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            fwFile = CharStreams.writingNonStringToFileFileWriter(TODO_LIST_FW_FILE, TODO_1);
+            assertThat(fwFile).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenPrintWriterAppendsToFile() throws IOException {
+        try {
+            pwFile = CharStreams.appendingToFilePrintWriter(TODO_LIST_PW_FILE, TODO_2);
+            assertThat(pwFile).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenFileWriterAppendsToFile() throws IOException {
+        try {
+            fwFile = CharStreams.appendingToFileFileWriter(TODO_LIST_FW_FILE, TODO_2);
+            assertThat(fwFile).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenCheckingError_thenPrintWriterReturnsFalse() throws IOException {
+        try {
+            boolean result = CharStreams.printWriterErrorFlag(TODO_LIST_FW_FILE, TODO_2);
+            assertFalse(result);
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+    
+
+    @AfterEach
+    public void clean() throws IOException {
+        Files.deleteIfExists(Paths.get(TODO_LIST_PW_FILE));
+        Files.deleteIfExists(Paths.get(TODO_LIST_FW_FILE));

```

Comment:
To avoid yet another conversion to/from strings and files, please use the existing file variables here:

```suggestion
        Files.deleteIfExists(pwFile.toPath());
        Files.deleteIfExists(fwFile.toPath());

```